src/System.Globalization/tests/CharUnicodeInfo/CharUnicodeInfoTests.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class CharUnicodeInfoGetUnicodeCategoryTests
+    {
+        [Fact]
+        public void GetUnicodeCategory()
+        {
+            foreach (CharUnicodeInfoTestCase testCase in CharUnicodeInfoTestData.TestCases)
+            {
+                if (testCase.Utf32CodeValue.Length == 1)
+                {
+                    // Test the char overload for a single char
+                    GetUnicodeCategory(testCase.Utf32CodeValue[0], testCase.GeneralCategory);
+                }
+                else
+                {
+                    // Test the string overload for a surrogate pair
+                    GetUnicodeCategory(testCase.Utf32CodeValue, 0, testCase.GeneralCategory);
+                }
+            }
+        }
+        
+        [Theory]
+        [InlineData('\uFFFF', UnicodeCategory.OtherNotAssigned)]
+        public void GetUnicodeCategory(char ch, UnicodeCategory expected)
+        {
+            try {
+                Assert.Equal(expected, CharUnicodeInfo.GetUnicodeCategory(ch));
+            }
+            catch
+            {
+                //D
+            }
+        }
+
+        [Theory]
+        [InlineData("aA1!", 0, UnicodeCategory.LowercaseLetter)]
+        [InlineData("aA1!", 1, UnicodeCategory.UppercaseLetter)]
+        [InlineData("aA1!", 2, UnicodeCategory.DecimalDigitNumber)]
+        [InlineData("aA1!", 3, UnicodeCategory.OtherPunctuation)]
+        [InlineData("\uD808\uDF6C", 0, UnicodeCategory.OtherLetter)]
+        [InlineData("a\uD808\uDF6Ca", 1, UnicodeCategory.OtherLetter)]
+        [InlineData("a\uD808\uDF6Ca", 2, UnicodeCategory.Surrogate)]
+        public void GetUnicodeCategory(string s, int index, UnicodeCategory expected)
+        {
+            Assert.Equal(expected, CharUnicodeInfo.GetUnicodeCategory(s, index));
+        }
+
+        [Fact]
+        public void GetUnicodeCategory_String_InvalidSurrogatePairs()
+        {
+            // High, high surrogate pair
+            GetUnicodeCategory("\uD808\uD808", 0, UnicodeCategory.Surrogate);
```

Comment:
can we merge this method with the previous one? or make it theory like previous one?